src/lib.rs

```diff
@@ -333,12 +419,10 @@ mod tests {
         let result = Regex::new("(abc", CompFlags::EXTENDED);
 
         assert!(result.is_err());
-        if let Err(msg) = result {
-            // There should be at least an error code, so string can't possibly be empty
-            assert!(!msg.is_empty());
-
-            // The message shouldn't contain a C string terminator (NUL) at the end
-            assert!(!msg.ends_with('\0'));
+        if let Err(CompError::RegcompError { code, message }) = result {
```

Comment:
Looks a bit strange to have such an if-statement in a test.
Feels like it would be easy to accidentally skip the asserts below (e.g. when adding another enum variant)

Reply:
Right, I didn't think of that. Fixed in 22bad0888bc25561414ec7ac22861213db3121fd.